calicoctl/calico_ctl/node.py

```diff
@@ -374,6 +390,83 @@ def node_start(node_image, runtime, log_dir, ip, ip6, as_num, detach,
                                         etcd_volumes, etcd_binds)
 
 
+def _ensure_host_tunnel_addr(ipv4_pools, ipip_pools):
+    """
+    Ensure the host has a valid IP address for its IPIP tunnel device.
+
+    This must be an IP address claimed from one of the IPIP pools.
+    Handles re-allocating the address if it finds an existing address
+    that is not from an IPIP pool.
+
+    :param ipv4_pools: List of all IPv4 pools.
+    :param ipip_pools: List of IPIP-enabled pools.
+    """
+    ip_addr = _get_host_tunnel_ip()
+    if ip_addr:
+        # Already have an IP address configured. Check that it's still
+        # within one of the IPIP pools.
+        for pool in ipv4_pools:
+            if ip_addr in pool.cidr:
+                # Found the right pool, check it still has IPIP enabled.
+                if not pool.ipip:
+                    # No longer an IPIP pool. Release the IP and allocate
+                    # a new one.
+                    client.release_ips({ip_addr})
+                    ip_addr = None
+                break
+        else:
+            # Not in any IPIP pool.  IP must be stale.  Since it's not in
+            # any pool, we can't release it.
+            ip_addr = None
+    if not ip_addr:
+        # Either there was no IP or the IP needs to be replaced.  Try to
+        # get an IP from one of the IPIP-enabled pools.
+        for ipip_pool in ipip_pools:
+            v4_addrs, _ = client.auto_assign_ips(
```

Comment:
That's a shame, I was hoping that auto_assign_ips would allow a list of pools to be passed in. No biggie though and probably not worth changing.